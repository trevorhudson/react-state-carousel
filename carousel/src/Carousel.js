import { useState } from "react";
import "./Carousel.css";
import Card from "./Card";


/** Carousel: displays images and arrows to navigate through them
 *
 * Props:
 * - photos: array of {src, caption} objects
 * - title: string describing the collection of images
 *
 * State:
 * - currCardIdx: integer for current card index
 *
 * App --> Carousel --> Card
 */
function Carousel({ photos, title }) {
  const [currCardIdx, setCurrCardIdx] = useState(0);

  console.log('Current Card', currCardIdx, setCurrCardIdx);
  const currCard = photos[currCardIdx];
  const total = photos.length;

  //Increments currCardIdx state by 1 -> loops at end.
  function goForward() {
    setCurrCardIdx(currCardIdx === total - 1 ? 0 : currCardIdx + 1);
  }

  //Decrements currCardIdx state by 1 -> loops at end.
  function goBackward() {
    setCurrCardIdx(currCardIdx === 0 ? total - 1 : currCardIdx - 1);
  }

  return (
    <div className="Carousel">
      <h1>{title}</h1>
      <div className="Carousel-main">

        <i
          className="bi bi-arrow-left-circle"
          onClick={goBackward}
          style={{ "visibility": currCardIdx === 0 ? "hidden" : "visible" }}
        />

        <Card
          caption={currCard.caption}
          src={currCard.src}
          currNum={currCardIdx + 1}
          totalNum={total}
        />
        <i
          className="bi bi-arrow-right-circle"
          onClick={goForward}
          style={{ "visibility": currCardIdx === 2 ? "hidden" : "visible" }}

        />
      </div>
    </div>
  );
}

export default Carousel;
