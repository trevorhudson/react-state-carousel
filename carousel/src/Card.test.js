import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";

describe("Tests for Card", function () {

  it("matches snapshot", function () {
    const testCard = TEST_IMAGES[0];

    const { container } = render(
      <Card
        caption={testCard.caption}
        src={testCard.src}
        currNum={0}
        tota
        lNum={TEST_IMAGES.length}
      />);

    expect(container).toMatchSnapshot();

  });
  it("renders Card without crashing", function () {
    const testCard = TEST_IMAGES[0];

    render(
      <Card
        caption={testCard.caption}
        src={testCard.src}
        currNum={0}
        totalNum={TEST_IMAGES.length}
      />);

  });


});