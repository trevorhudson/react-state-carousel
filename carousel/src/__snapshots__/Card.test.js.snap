// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Card matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 
      0
       of 
      .
    </small>
  </div>
</div>
`;
